iothub_client/src/blob.c

```diff
@@ -219,92 +219,86 @@ BLOB_RESULT Blob_UploadMultipleBlocksFromSasUri(const char* SASURI, IOTHUB_CLIEN
                                     /*Codes_SRS_BLOB_02_021: [ For every block returned by `getDataCallback` the following operations shall happen: ]*/
                                     unsigned int blockID = 0; /* incremented for each new block */
                                     unsigned int isError = 0; /* set to 1 if a block upload fails or if getDataCallback returns incorrect blocks to upload */
-                                    unsigned int uploadOneMoreBlock; /* set to 1 while getDataCallback returns correct blocks to upload */
+                                    unsigned int uploadOneMoreBlock = 1; /* set to 1 while getDataCallback returns correct blocks to upload */
                                     unsigned char const * source; /* data set by getDataCallback */
                                     size_t size; /* source size set by getDataCallback */
+                                    IOTHUB_CLIENT_FILE_UPLOAD_GET_DATA_RESULT getDataReturnValue;
 
-                                    /* get first block */
-                                    getDataCallback(FILE_UPLOAD_OK, &source, &size, context);
+                                    result = BLOB_OK; /* Initialization needed here in case no block is uploaded, which is be fine */
```

Comment:
We have a guideline not to assign a value to a return var upfront unless really needed (example: in a function with a logic complex enough to get a compiler confused and think we are not assigning a value to the return var, which causes compilation warnings and errors).
*If possible*, please get the var value assigned only at the specific decision block that justifies that state.

Reply:
Yes, it is possible. Good catch. I refactored a little bit. I hope the code is better and clearer now.